commcare_connect/opportunity/views.py

```diff
@@ -512,14 +512,15 @@ def export_users_for_payment(request, org_slug, pk):
 def payment_import(request, org_slug=None, pk=None):
     opportunity = get_opportunity_or_404(org_slug=org_slug, pk=pk)
     file = request.FILES.get("payments")
-    try:
-        status = bulk_update_payment_status(opportunity, file)
-    except ImportException as e:
-        messages.error(request, e.message)
-    else:
-        message = f"Payment status updated successfully for {len(status)} users."
-        messages.success(request, mark_safe(message))
-    return redirect("opportunity:detail", org_slug, pk)
+
+    file_format = get_file_extension(file)
+    if file_format not in ("csv", "xlsx"):
+        raise ImportException(f"Invalid file format. Only 'CSV' and 'XLSX' are supported. Got {file_format}")
+    imported_data = get_imported_dataset(file, file_format)
+    rows = list(imported_data)
```

Comment:
If the import is quite large this will take a lot of memory in the workers and redis (our celery broker). It would be good to add some kind of length limit here, since it is so easy to break the sheet into multiple, and ideally add locking so that only one task per opportunity can be run. I wouldn't expect payments to hit the limit (unlike visit import), but better to be safe. Alternatively, you can save the sheet in s3, and pass a key to the celery task so that we don't need to store the info in redis at all.

Reply:
Implemented storing it as file and passing ref.